compilation/platforms/solc.go

```diff
@@ -104,6 +104,18 @@ func (s *SolcCompilationConfig) Compile() ([]types.Compilation, string, error) {
 
 	// Create a compilation unit out of this.
 	compilation := types.NewCompilation()
+	if sourceList, ok := results["sourceList"]; ok {
+		if sourceListCasted, ok := sourceList.([]any); ok {
```

Comment:
Why don't we assert `sourceList.([]string)`? In that way the errors make more sense and we don't have to "worry" about `sourceListCasted[i].(string)` causing some unlikely panic.

Reply:
You can't do this because of how JSON marshaling and Go types handle these sorts of things. Although the elements of this `[]any` slice are of type `string`, the overarching array type is defined in Go as `[]any` by the `json` package. You can't cast a `[]any` to `[]string` in Go. We have a helper util for this I believe, but you still need to cast to `[]any` first.

We could add a check on the string casting though. Realistically it would be better to rewrite all of this to unmarshal structs in the future though, like the crytic platform code.

Reply:
Smh golang...thank you for the explanation.